pkgs/applications/version-management/datalad/default.nix

```diff
@@ -1,69 +1,91 @@
-{ lib, stdenv, fetchFromGitHub, installShellFiles, python3, git, git-annex }:
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  installShellFiles,
+  python3,
+  git,
+  git-annex,
+  p7zip,
+  curl,
+  coreutils,
+}:
 
 python3.pkgs.buildPythonApplication rec {
   pname = "datalad";
-  version = "1.0.2";
+  version = "1.1.1";
 
   src = fetchFromGitHub {
     owner = "datalad";
     repo = pname;
     rev = "refs/tags/${version}";
-    hash = "sha256-oq+DdlWcwjJSQdnqHlYCa9I7iSOKf+hI35Lcv/GM24c=";
+    hash = "sha256-Vw/RpMf+jnUijJ3GZ9nLk1IRWOADmM+jNtYl5Ba6uLg=";
   };
 
-  nativeBuildInputs = [ installShellFiles git ];
-
-  propagatedBuildInputs = with python3.pkgs; [
-    # core
-    platformdirs
-    chardet
-    iso8601
-    humanize
-    fasteners
-    packaging
-    patool
-    tqdm
-    annexremote
-    looseversion
-    setuptools
-    git-annex
-
-    # downloaders-extra
-    # requests-ftp # not in nixpkgs yet
-
-    # downloaders
-    boto
-    keyrings-alt
-    keyring
-    msgpack
-    requests
-
-    # publish
-    python-gitlab
-
-    # misc
-    argcomplete
-    pyperclip
-    python-dateutil
-
-    # metadata
-    simplejson
-    whoosh
-
-    # metadata-extra
-    pyyaml
-    mutagen
-    exifread
-    python-xmp-toolkit
-    pillow
-
-    # duecredit
-    duecredit
+  postPatch = ''
+    substituteInPlace datalad/distribution/create_sibling.py \
+      --replace-fail "/bin/ls" "${coreutils}/bin/ls"
+  '';
 
-    # python>=3.8
-    distro
-  ] ++ lib.optionals stdenv.hostPlatform.isWindows [ colorama ]
-    ++ lib.optionals (python3.pythonOlder "3.10") [ importlib-metadata ];
+  nativeBuildInputs = [
+    installShellFiles
+    git
+  ];
+
+  build-system = [ python3.pkgs.setuptools ];
+
+  dependencies =
+    with python3.pkgs;
+    [
+      # core
+      platformdirs
+      chardet
+      iso8601
+      humanize
+      fasteners
+      packaging
+      patool
+      tqdm
+      annexremote
+      looseversion
+      setuptools
+      git-annex
+
+      # downloaders-extra
+      # requests-ftp # not in nixpkgs yet
+
+      # downloaders
+      boto3
+      keyrings-alt
+      keyring
+      msgpack
+      requests
+
+      # publish
+      python-gitlab
+
+      # misc
+      argcomplete
+      pyperclip
+      python-dateutil
+
+      # metadata-extra
+      pyyaml
+      mutagen
+      exifread
+      python-xmp-toolkit
+      pillow
```

Comment:
You add some packages, but for example pyyaml, mutagen and others do not exist in `setup.py`. You need to remove them.

Reply:
Oh yes, I must've overlooked them. Resolved.